aea/manager.py

```diff
@@ -310,8 +311,12 @@ def add_project(
         :param local: whether or not to fetch from local registry.
         :param restore: bool flag for restoring already fetched agent.
         """
-        if public_id in self._projects:
-            raise ValueError(f"Project {public_id} was already added!")
+        if public_id.to_latest() in self._projects_set:
```

Comment:
You should do a prefix check; this now checks whether `autor/name:latest` is in the set. What we should check is if `autor/name` is in the set!

Reply:
that correct, instead of forming prefix manually I use to_latest() to get rid of actual versions.
original version info is stored in self._projects